pointer-masking-proposal.adoc

```diff
@@ -295,76 +298,133 @@ MASK
 Figure 6d: Memory Tagging Extension register (_**upmmask**_) for RV128 (U-MODE) +
  +
 
+[%header, cols=1*]
+,===
+mpmbase[31:0]
+BASE
+,===
+Figure 7a: Memory Tagging Extension register (_**mpmbase**_) for RV32 (M-MODE) +
+ +
+
+[%header, cols=1*]
+,===
+hpmbase[31:0]
+BASE
+,===
+Figure 7b: Memory Tagging Extension register (_**hpmbase**_) for RV32 (H-MODE) +
+ +
+
 [%header, cols=1*]
 ,===
 spmbase[31:0]
 BASE
 ,===
-Figure 7a: Memory Tagging Extension register (_**spmbase**_) for RV32 (S-MODE) +
+Figure 7c: Memory Tagging Extension register (_**spmbase**_) for RV32 (S-MODE) +
  +
 
 [%header, cols=1*]
 ,===
 upmbase[31:0]
 BASE
 ,===
-Figure 7b: Memory Tagging Extension register (_**upmbase**_) for RV32 (U-MODE) +
+Figure 7d: Memory Tagging Extension register (_**upmbase**_) for RV32 (U-MODE) +
+ +
+
+[%header, cols=1*]
+,===
+mpmbase[63:0]
+BASE
+,===
+Figure 8a: Memory Tagging Extension register (_**mpmbase**_) for RV64 (M-MODE) +
+ +
+
+[%header, cols=1*]
+,===
+hpmbase[63:0]
+BASE
+,===
+Figure 8b: Memory Tagging Extension register (_**hpmbase**_) for RV64 (H-MODE) +
  +
 
 [%header, cols=1*]
 ,===
 spmbase[63:0]
 BASE
 ,===
-Figure 8a: Memory Tagging Extension register (_**spmbase**_) for RV64 (S-MODE) +
+Figure 8c: Memory Tagging Extension register (_**spmbase**_) for RV64 (S-MODE) +
  +
 
 [%header, cols=1*]
 ,===
 upmbase[63:0]
 BASE
 ,===
-Figure 8b: Memory Tagging Extension register (_**upmbase**_) for RV64 (U-MODE) +
+Figure 8d: Memory Tagging Extension register (_**upmbase**_) for RV64 (U-MODE) +
+ +
+
+[%header, cols=1*]
+,===
+mpmbase[127:0]
+BASE
+,===
+Figure 9a: Memory Tagging Extension register (_**mpmbase**_) for RV128 (M-MODE) +
+ +
+
+[%header, cols=1*]
+,===
+hpmbase[127:0]
+BASE
+,===
+Figure 9b: Memory Tagging Extension register (_**hpmbase**_) for RV128 (H-MODE) +
  +
 
 [%header, cols=1*]
 ,===
 spmbase[127:0]
 BASE
 ,===
-Figure 9a: Memory Tagging Extension register (_**spmbase**_) for RV128 (S-MODE) +
+Figure 9c: Memory Tagging Extension register (_**spmbase**_) for RV128 (S-MODE) +
  +
 
 [%header, cols=1*]
 ,===
 upmbase[127:0]
 BASE
 ,===
-Figure 9b: Memory Tagging Extension register (_**upmbase**_) for RV128 (U-MODE) +
+Figure 9d: Memory Tagging Extension register (_**upmbase**_) for RV128 (U-MODE) +
  +
 
 [[explanation]]
 ## Explanation
 
-_**MMTE**_ register fully two-fold function:
+_**MMTE**_ register fulfills two-fold function:
 
-1. Can be programmable by more privileged mode
+1. Can be programmable by more privileged mode (unless PM.Current bit is enabled)
```

Comment:
Which registers are affected with PM.Current set to 1: only xPMMASK and xPMBASE or also MMTE? There're some contradicting lines in the spec.

Reply:
If PM.Current=1, current privilege level has rights to self-manage PM.Enable bit as well as modify xPMMASK and xPMBASE. This allows to self-manage in-process isolation feature.